arwen/contexts/runtime.go

```diff
@@ -66,17 +72,36 @@ func (context *runtimeContext) InitStateFromContractCallInput(input *vmcommon.Co
 
 func (context *runtimeContext) PushState() {
 	newState := &runtimeContext{
-		vmInput:      context.vmInput,
-		scAddress:    context.scAddress,
-		callFunction: context.callFunction,
-		readOnly:     context.readOnly,
+		vmInput:       context.vmInput,
+		scAddress:     context.scAddress,
+		callFunction:  context.callFunction,
+		readOnly:      context.readOnly,
+		asyncCallInfo: context.asyncCallInfo,
 	}
 
 	context.stateStack = append(context.stateStack, newState)
```

Comment:
let's add a limit of 10 states at most. 10 to be configurable.

Reply:
Indeed. Will come as a separate PR, because that change would require this method to return errors which need handling, and it would be messy in this PR. But the limit is required and will be added.